pkg/testframework/console-operator.go

```diff
@@ -210,3 +210,51 @@ func SetLogLevel(t *testing.T, client *Clientset, logLevel operatorsv1.LogLevel)
 func GenerationChanged(oldGeneration, newGeneration int64) bool {
 	return oldGeneration == newGeneration
 }
+
+// the operator is settled if conditions match:
+// - available: true
+// - progressing: false
+// - degraded: false
+func operatorIsSettled(operatorConfig *operatorsv1.Console) bool {
+	settled := true
+	for _, condition := range operatorConfig.Status.Conditions {
+		if condition.Type == operatorsv1.OperatorStatusTypeAvailable {
+			if condition.Status == operatorsv1.ConditionFalse {
+				settled = false
+			}
+		}
+		if condition.Type == operatorsv1.OperatorStatusTypeProgressing {
+			if condition.Status == operatorsv1.ConditionTrue {
+				settled = false
+			}
+		}
+		if condition.Type == operatorsv1.OperatorStatusTypeDegraded {
+			if condition.Status == operatorsv1.ConditionTrue {
+				settled = false
+			}
+		}
+	}
+	return settled
+}
+
+// A helper to ensure our operator config reaches a settled state before we
+// begin the next test.
+func WaitForSettledState(t *testing.T, client *Clientset) (settled bool, err error) {
+	interval := 1 * time.Second
+	max := 20 * time.Second
+	count := 0
+	pollErr := wait.Poll(interval, max, func() (stop bool, err error) {
+		count++
+		operatorConfig, err := client.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return operatorIsSettled(operatorConfig), nil
+	})
+	if pollErr != nil {
+		t.Errorf("operator did not reach settled state: %v", pollErr)
```

Comment:
Not a native, but maybe ?
```suggestion
		t.Errorf("operator has not reached settled state: %v", pollErr)
```

Reply:
If you call `t.Errorf(...)`, or any of the Error* methods from the testing pkg, the test fail instantly.
Not sure the `return` below will make any difference.
If anything would get rid of the `t.Errorf(...)`

Reply:
Will update!

I was contemplating if this should definitely fail the test. I think so, otherwise, the NEXT test may fail, and tracking down the error would be harder.